### PR TITLE
Handling invalid host name instead of throwing NPE #123

### PR DESCRIPTION
**Pull request by druminski**

### CODE REVIEW

File: hermes-api/src/main/java/pl/allegro/tech/hermes/api/constraints/EndpointAddressValidator.java [hunk 1]
```diff
@@ -44,9 +55,16 @@ private boolean validateWithTemplate(String address, ConstraintValidatorContext
...
 
-    protected void createConstraintMessage(ConstraintValidatorContext context, String message) {
+    private void createConstraintMessage(ConstraintValidatorContext context, String message) {
         context.buildConstraintViolationWithTemplate(message)
                 .addConstraintViolation()
                 .disableDefaultConstraintViolation();
     }
+
+    private boolean isInvalidHost(UriTemplate template) {
+        Map<String, Object> uriKeysWithEmptyValues = Arrays.asList(template.getVariables()).stream().collect(toMap(v -> v, v -> "empty"));
```
Comment (wojtkiewicz) [inline hermes-api/src/main/java/pl/allegro/tech/hermes/api/constraints/EndpointAddressValidator.java]: You can use Function.identity() to express function that returns input arguments.

**Comment by adamdubiel** [comment]
+1

File: hermes-api/src/main/java/pl/allegro/tech/hermes/api/constraints/EndpointAddressValidator.java [hunk 2]
```diff
@@ -44,9 +55,16 @@ private boolean validateWithTemplate(String address, ConstraintValidatorContext
...
+
+        //check if host is null due to bug in jdk http://bugs.java.com/view_bug.do?bug_id=5083594
+        return URI.create(template.expand(uriKeysWithEmptyValues)).getHost() == null;
```
Comment (pjagielski) [inline hermes-api/src/main/java/pl/allegro/tech/hermes/api/constraints/EndpointAddressValidator.java]: I think it's not a good solution to propagate this JDK behaviour into our code. If it doesn't work we should use something different to validate the uri.

File: hermes-api/src/main/java/pl/allegro/tech/hermes/api/constraints/EndpointAddressValidator.java (hunk 1 above)
Comment (rzukow) [inline hermes-api/src/main/java/pl/allegro/tech/hermes/api/constraints/EndpointAddressValidator.java]: Also, there is a nice Arrays.stream method

File: hermes-api/src/main/java/pl/allegro/tech/hermes/api/constraints/EndpointAddressValidator.java (hunk 2 above)
Comment (druminski) [inline hermes-api/src/main/java/pl/allegro/tech/hermes/api/constraints/EndpointAddressValidator.java]: Yes, we could try to use other library but I don't want to add to this module other dependencies. Hermes-api should be a thin module.

File: hermes-api/src/main/java/pl/allegro/tech/hermes/api/constraints/EndpointAddressValidator.java (hunk 1 above)
Comment (wojtkiewicz) [inline hermes-api/src/main/java/pl/allegro/tech/hermes/api/constraints/EndpointAddressValidator.java]: +1

Comment (druminski) [inline hermes-api/src/main/java/pl/allegro/tech/hermes/api/constraints/EndpointAddressValidator.java]: done

**Comment by wojtkiewicz** [comment]
:+1:

**Pull request merged**